python/smqtk/representation/descriptor_element/postgres.py

```diff
@@ -232,22 +216,11 @@ def has_vector(self):
             "uuid_val": str(self.uuid())
         }
 
-        conn = self._get_psql_connection()
-        cur = conn.cursor()
-
-        try:
-            self._ensure_table(cur)
+        def exec_hook(cur):
             cur.execute(q_select, q_select_values)
-            r = cur.fetchone()
-            # For server cleaning (e.g. pgbouncer)
-            conn.commit()
-            return bool(r)
-        except:
-            conn.rollback()
-            raise
-        finally:
-            cur.close()
-            conn.close()
+
+        # Should either yield one or zero rows
+        return bool(list(self.psql_helper.single_execute(exec_hook, True)))
```

Comment:
Explicitly mark defaulted keyword arguments, e.g. ``...single_execute(exec_hook, yield_result_rows=True)`` to protect from key-word argument reordering and future added positional arguments.